src/components/ClosePurchaseOrderModal.vue

```diff
@@ -130,20 +131,30 @@ export default defineComponent({
       }
 
       const eligibleItems = this.order.items.filter((item: any) => item.isChecked && this.isPOItemStatusPending(item))
-      const responses = await Promise.allSettled(eligibleItems.map(async (item: any) => {
-        await OrderService.updatePOItemStatus({
-          orderId: item.orderId,
-          orderItemSeqId: item.orderItemSeqId,
-          statusId: "ITEM_COMPLETED"
-        })
-        return item.orderItemSeqId
-      }))
-      const failedItemsCount = responses.filter((response) => response.status === 'rejected').length
-      if(failedItemsCount){
-        console.error('Failed to update the status of purchase order items.')
+      let hasFailedItems = false;
+      let completedItems = [] as any;
+
+      for(const item of eligibleItems) {
```

Comment:
Instead of making synchronous calls, we can update items in batches. Let's say there are N items to update. We can process N−1 items in batches, and then update the final item synchronously after all batch processing is complete. This way PO status get updated correctly after all items are completed. 

Later we can think of some api or csv data import mechanism to update order item status in bulk.